### PR TITLE
[IVY-2910] Fix Incorrect Balance Transfer

### LINKED ISSUE #2910: [BUG] Incorrect addition on account transfer
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate problems

### Describe the bug

One of my accounts had a balance of 32.45€. I transferred 167.55€ from another account to make it 200€. Adding this on ivy wallet somehow results in ivy showing a balance of 199€ in the first account.

I modified the amount of the account transfer on ivy wallet to observe some funny results, pfa some screenshots. The same error is also there if I change the amount to 67,55 (the total should be 100€ but it shows 99€)

### To Reproduce

Add an account transfer to make the balance a multiple of 100

### Expected behavior

The new balance should be exact

### Screenshots

Screenshot_2024-02-02-10-23-50-84_5e9f1c73c4ea0ad7b9937a61f977427e
Screenshot_2024-02-02-10-23-40-33_5e9f1c73c4ea0ad7b9937a61f977427e
Screenshot_2024-02-02-10-16-43-13_5e9f1c73c4ea0ad7b9937a61f977427e
Screenshot_2024-02-02-10-16-28-93_5e9f1c73c4ea0ad7b9937a61f977427e
Screenshot_2024-02-02-10-16-14-58_5e9f1c73c4ea0ad7b9937a61f977427e
Screenshot_2024-02-02-10-16-07-47_5e9f1c73c4ea0ad7b9937a61f977427e

### Smartphone

OnePlus Nord CE 2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Vishwa-Raghavendra**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] 🎬 I've attached a **screen recoding** of the changes. 

> Tip: drag & drop the video to the PR description.

## What's changed?

Describe with a few bullets **what's new:**
- a This fixes In correct Account Transfer that was happening (this isn't a rework)
- b
- c

> 💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!

## Risk Factors

**What may go wrong if we merge your PR?**

- a
- b

**In what cases your code won't work?**

- a
- b

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Closes #2910 

> Replace `ISSUE_NUMBER` with the number of the GitHub issue that this PR is fixing. If you've done that correctly, you'll see the issue title linked when previewing your PR description.

> **!Note:** Do **not** link the PR number. Link the number/id of the GitHub Issue from [issues](https://github.com/Ivy-Apps/ivy-wallet/issues).

## Troubleshooting CI failures ❌

GitHub Actions failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

File: temp/legacy-code/src/main/java/com/ivy/legacy/domain/pure/transaction/AccountValueFunctions.kt [hunk 1]
```diff
@@ -52,7 +52,7 @@ object AccountValueFunctions {
         transaction: Transaction,
         accountId: UUID
     ): BigDecimal = with(transaction) {
-        if (this.getAccountId() == accountId && this is Transfer) {
+        if (this is Transfer && this.toAccount.value == accountId) {
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/domain/pure/transaction/AccountValueFunctions.kt:55]: Shouldn't we do the same for `transferExpense`? Also can you please explain the logic behind this

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
Do we need the same fix for `transferExpense`?

**Comment by Vishwa-Raghavendra** [comment]
> Do we need the same fix for `transferExpense`?

Not Needed for TransfersExpense

File: temp/legacy-code/src/main/java/com/ivy/legacy/domain/pure/transaction/AccountValueFunctions.kt (hunk 1 above)
Comment (Vishwa-Raghavendra) [reply temp/legacy-code/src/main/java/com/ivy/legacy/domain/pure/transaction/AccountValueFunctions.kt:55]: The logic is simple 

If the toAccountId of a transaction match with the current accountId in consideration then it is treated as income for that account 

For any Transaction 

X --> Y 

The amount will always be an income for Y account and expense for X account

**Review by ILIYANGERMANOV — APPROVED** [review]

**Comment by ILIYANGERMANOV** [comment]
> > Do we need the same fix for `transferExpense`?
> 
> Not Needed for TransfersExpense

@Vishwa-Raghavendra I don't get it. Why we need it only for income and not for expense? That feels incorrect

**Comment by Vishwa-Raghavendra** [comment]
> > > Do we need the same fix for `transferExpense`?
> > 
> > 
> > Not Needed for TransfersExpense
> 
> @Vishwa-Raghavendra I don't get it. Why we need it only for income and not for expense? That feels incorrect

I meant we don't need any changes 
the existing code is already checking for transferExpense 

i.ie For a transfer from X to Y 
is the fromX.accountId == accountId in consideration 

Only for transferIncome it was broken so just fixed it up

**Comment by ILIYANGERMANOV** [comment]
Got it - thanks!

**Pull request merged**